common/metrics/runtime.go

```diff
@@ -44,8 +44,8 @@ const (
 
 // RuntimeMetricsReporter A struct containing the state of the RuntimeMetricsReporter.
 type RuntimeMetricsReporter struct {
-	provider          MetricsHandler
-	buildInfoProvider MetricsHandler
+	provider          Handler
+	buildInfoProvider Handler
```

Comment:
```suggestion
	handler          Handler
	buildInfoHandler Handler
```
and again, check all other places.